src/renderers/dom/shared/eventPlugins/SimpleEventPlugin.js

```diff
@@ -126,14 +133,21 @@ var topLevelEventsToDispatchConfig: {[key: TopLevelTypes]: DispatchConfig} = {};
   var capitalizedEvent = event[0].toUpperCase() + event.slice(1);
   var onEvent = 'on' + capitalizedEvent;
   var topEvent = 'top' + capitalizedEvent;
+  var isPhased = !topLevelDirectDispatches[topEvent];
 
   var type = {
-    phasedRegistrationNames: {
-      bubbled: onEvent,
-      captured: onEvent + 'Capture',
-    },
     dependencies: [topEvent],
   };
+
+  if (isPhased) {
+    type.phasedRegistrationNames = {
+      bubbled: onEvent,
+      captured: onEvent + 'Capture',
+    };
+  } else {
+    type.registrationName = onEvent;
+  }
```

Comment:
This makes the `type` object polymorphic which could hurt performance a bit. We may want to define these fields on `type` with null values to avoid creating multiple hidden classes.